spaceship/lib/spaceship/tunes/availability.rb

```diff
@@ -51,16 +68,50 @@ def self.from_territories(territories = [], params = {})
         obj.include_future_territories = params.fetch(:include_future_territories, true)
         obj.cleared_for_preorder = params.fetch(:cleared_for_preorder, false)
         obj.app_available_date = params.fetch(:app_available_date, nil)
+        obj.b2b_unavailable =  params.fetch(:b2b_unavailable, false)
+        obj.b2b_app_enabled =  params.fetch(:b2b_app_enabled, false)
+        obj.educational_discount = params.fetch(:educational_discount, true)
         return obj
       end
 
       def territories
         @territories ||= raw_data['countries'].map { |info| Territory.new(info) }
       end
 
+      def b2b_users
+        @b2b_users ||= raw_data['b2bUsers'].map { |user| B2bUser.new(user) }
+      end
+
+      def b2b_app_enabled
+        @b2b_app_enabled.nil? ? raw_data['b2bAppEnabled'] : @b2b_app_enabled
+      end
+
+      def educational_discount
+        @educational_discount.nil? ? raw_data['educationalDiscount'] : @educational_discount
+      end
+
       def cleared_for_preorder
         super || false
       end
+
+      # Sets the b2b flag. If you call Save on app_details without adding any b2b users
+      # it will result in an error.
+      def enable_b2b_app!
+        raise "Not possible to enable b2b on this app" if b2b_unavailable
+        @b2b_app_enabled = true
+        # need to set the educational discount to false
+        @educational_discount = false
+        self
+      end
+
+      # just adds users to the availability, You will still have to call update_availabilty
+      def add_b2b_users(user_list = [])
```

Comment:
Does this actually "set" the users and not "add" users? 🤔

Reply:
If this does set/override the `@b2b_users`, could we maybe combine `enable_b2b_app!` with this one and just pass the user list into the `enable_b2_app!` method?

Reply:
It does not set or override. It may look like this but the way the requests are ( if you observe the object ) only the email ids either being added or removed are sent over.

Reply:
@heartofagoof Interesting 🤔 Good to know! So this _could_ also remove the user if the user was already set?

Reply:
No. For that you'd have to set removed = true. Which we are not setting right now. I didn't want to implement anything destructive. :)

Reply:
I can of course change the implementation to something like this if it helps : 

`update_b2b_users(users_to_add: [], users_to_remove: [])`

and update the signature to: 

`from_username(email_id: [], add_or_remove:)`

Reply:
This way we provide the ability to add and remove b2b teams.